scrunch/datasets.py

```diff
@@ -529,8 +529,8 @@ def _reload_variables(self):
         self.order = DatasetVariablesOrder(self._catalog, order)
 
     def __iter__(self):
-        for var in self._vars:
-            yield var
+        for subvar in self.resource['body']['subvariables']:
```

Comment:
this line will make scrunch fail, this method is used for handling Variables of a Dataset and Subvariables of a Variable, so in the case of Variables it wil fail with a KeyError. Dataset has no key subvariables.